### PR TITLE
Removes support for implicitly determined view properties in bindings

### LINKED ISSUE #846: Remove OneWayBind implicit resolution of view property type
There are several related issues here:

1. *Reflection.cs* utilizes `GetRuntimeField` and `GetRuntimeProperty` in its `getViewExpression` method. The behavior of these methods has changed in FX4.6 such that they no longer return private members.
2. *PropertyBinding.cs* uses `GetDeclaredMethod` in its `OneWayBind<TViewModel, TView, TVMProp, TVProp>` method. However, there are multiple matches for the passed in method, so it will throw due to the ambiguity.
3. In the same method as (2), there is a check to see whether `tvProp != typeof(TVProp)`. If not, a dynamic method invocation and cast is performed. However, the cast can never succeed if `tvProp != typeof(TVProp)`.

**UPDATE**: after discussing this on Slack, it seems that it is probably best to just remove implicit view property resolution altogether. It results in opaque, more fragile, less performant code. Oh, and it probably hasn't worked for a long time anyway.

### PR DESCRIPTION
**Pull request by kentcb**
This PR fixes #846 by removing `IDefaultPropertyBinderProvider` and everything that depends on it, including the overloads of `Bind`, `OneWayBind`, and `BindCommand`.

Note that I have fixed up project files for all platforms, but am unable to verify all builds.

### CODE REVIEW

File: CommonAssemblyInfo.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
 using System.Reflection;
 
-[assembly: AssemblyVersion("6.5.0")]
-[assembly: AssemblyFileVersion("6.5.0")]
+[assembly: AssemblyVersion("6.90.0")]
```
Comment (jlaanstra) [inline CommonAssemblyInfo.cs]: Can you revert this file? It is not really part of the change made here.

Comment (jlaanstra) [inline CommonAssemblyInfo.cs]: Nevermind noticed that it comes from the merge. Can you retarget this to rxui7-master?

Comment (kentcb) [inline CommonAssemblyInfo.cs]: Hmm, I thought I did that, but maybe I did something wrong. How exactly should I do so?

Comment (jlaanstra) [inline CommonAssemblyInfo.cs]: Before clicking create it should give you the option to switch the target branch.

Comment (kentcb) [inline CommonAssemblyInfo.cs]: Ah, I misunderstood what Paul was asking. OK, should be good now but I still see lots of commits that aren't mine.
